### ISSUE TITLE
Trainer.save_state does not work if update_on_kvstore=False

### ISSUE THREAD

**Issue by eric-haibin-lin** [apache/incubator-mxnet#15918]
When update_on_kvstore is set to False and multiple GPUs are used, trainer only saves the trainer state on context[0], and the rest are not saved.

**Comment by ptrendx** [comment]
Shouldn't those states be exactly the same for all contexts?

**Comment by eric-haibin-lin** [comment]
Yes. `load_states` only loads the state for context[0] if update_on_kvstore=False, which makes GPU memory explode..
